### PR TITLE
HIP 206 create functions test coverage

### LINKED ISSUE #107: Complete HIP 206 create functions test coverage
### Problem

In teh initial coverage of HIP 206 functions from [TokenCreate](https://github.com/hashgraph/hedera-smart-contracts/blob/main/contracts/hts-precompile/TokenCreate.sol) were tests in [TokenCreateContract.js](https://github.com/hashgraph/hedera-smart-contracts/blob/main/test/hts-precompile/token-create/tokenCreateContract.js)
However, the coverage is incomplete and would serve as a good reference as to how DApp developers using ethersjs might utilize the create methods from IHederaTokenService.sol

### Solution

Depends on https://github.com/hashgraph/hedera-smart-contracts/issues/109

Complete test cases coverage

Update [TokenCreateContract.js](https://github.com/hashgraph/hedera-smart-contracts/blob/main/test/hts-precompile/token-create/tokenCreateContract.js) with simple tests using ethers to verify
- [ ] `createFungibleTokenPublic`
- [ ] `createNonFungibleTokenPublic`
- [ ] [createFungibleTokenWithCustomFeesPublic](https://github.com/hashgraph/hedera-smart-contracts/blob/main/contracts/hts-precompile/examples/token-create/TokenCreateContract.sol#L111)
- [ ] `createNonFungibleTokenWithCustomFeesPublic`
- [ ] `mintTokenPublic` - this seems to already exist in [utils.mintNFT](https://github.com/hashgraph/hedera-smart-contracts/blob/main/test/hts-precompile/utils.js#L58) however it would be good placed in the test file as an actual test for reference
- [ ] `grantTokenKycPublic` - this seems to already exist. in [utils.grantTokenKyc](https://github.com/hashgraph/hedera-smart-contracts/blob/main/test/hts-precompile/utils.js#L78) however it would be good placed in the test file as an actual test for reference
- [ ] `getNonFungibleTokenInfoPublic`
- [ ] `getTokenInfoPublic`

Note: It is not necessary to cover every edge case. This should be simple cases to verify calls can be made .
In a sense it also showcases how DApps might use ethers to make these calls

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by Ivo-Yankov**
**Description**:
Adds tests for:
- createFungibleTokenPublic
- createNonFungibleTokenPublic
- createFungibleTokenWithCustomFeesPublic
- createNonFungibleTokenWithCustomFeesPublic
- mintTokenPublic
- grantTokenKycPublic
- getNonFungibleTokenInfoPublic
- getTokenInfoPublic

**Related issue(s)**:

Fixes #107 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: test/hts-precompile/token-create/tokenCreateContract.js [hunk 1]
```diff
@@ -91,4 +90,72 @@ describe("TokenCreateContract tests", function () {
...
+    const { responseCode } = receipt.events.filter(e => e.event === 'ResponseCode')[0].args;
+    expect(responseCode).to.equal(22);
+    const { serialNumbers } = receipt.events.filter(e => e.event === 'MintedToken')[0].args;
+    expect(serialNumbers[0].toNumber()).to.be.greaterThan(0);
+  });
+
+  it('should be able to execute grantTokenKyc', async function () {
+    const grantKycTx = await tokenCreateContract.grantTokenKycPublic(tokenAddress, signers[1].address, { gasLimit: 1000000 });
+    expect((await grantKycTx.wait()).events.filter(e => e.event === 'ResponseCode')[0].args.responseCode).to.equal(22);
+  });
```
Comment (Nana-EC) [inline test/hts-precompile/token-create/tokenCreateContract.js:160]: Could you add `unfreeze()` and `unpause()` from the contract

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Looks good, similar comments in other PR but won't repeat them here.
Please add `unfreeze()` and `unpause()` - I forgot to mention them in the ticket but they are part of the contract

File: test/hts-precompile/token-create/tokenCreateContract.js (hunk 1 above)
Comment (Ivo-Yankov) [reply test/hts-precompile/token-create/tokenCreateContract.js:160]: I moved those methods to `TokenManagementContract` since they and `freeze()` and `pause()` are all part of the same HIP. It didn't really make sence for those methods to be separated in different contracts.

Comment (Nana-EC) [reply test/hts-precompile/token-create/tokenCreateContract.js:160]: Sure, the original intention noted in the issues was
token create - any actions does to create and enable token transfers
token transfer - transfer operations
token manage - token and token account modifications beyond enablement
token query - token info retrieval

The `unfreeze`, `unpause` and `kycGranted` are debatable so i'm fine with what you have.

**Review by Nana-EC — APPROVED** [review]
